api/pkg/authz/enforcer/enforcer.go

```diff
@@ -114,244 +71,207 @@ func newEnforcer(
-		Body:   input,
-		Flavor: string(e.flavor),
-	}
-	res, err := e.ketoClient.UpsertOryAccessControlPolicyRole(params.WithTimeout(e.timeout))
+func (e *enforcer) GetRoleMembers(ctx context.Context, role string) ([]string, error) {
+	expandedRole, _, err := e.ketoReadClient.PermissionApi.ExpandPermissions(ctx).
+		Namespace("Role").
+		Relation("member").
+		Object(role).
+		MaxDepth(2).Execute()
 	if err != nil {
 		return nil, err
 	}
-	return &types.Role{
-		ID:      res.GetPayload().ID,
-		Members: res.GetPayload().Members,
-	}, nil
-}
-
-// CreatePolicy create a policy to allow subject do an operation against the specified resource
-func (e *enforcer) UpsertPolicy(
-	policyName string,
-	roles []string,
-	users []string,
-	resources []string,
-	actions []string,
-) (*types.Policy, error) {
-	fmtPolicy := e.formatPolicy(policyName)
-
-	fmtResources := make([]string, 0)
-	for _, res := range resources {
-		fmtResources = append(fmtResources, e.formatResource(res))
+	members := make([]string, 0)
+	for _, child := range expandedRole.GetChildren() {
+		members = append(members, child.Tuple.SubjectSet.Object)
 	}
 
-	fmtRoles := make([]string, 0)
-	for _, role := range roles {
-		fmtRoles = append(fmtRoles, e.formatRole(role))
-	}
+	return members, nil
+}
 
-	fmtUsers := make([]string, 0)
-	for _, user := range users {
-		fmtUsers = append(fmtUsers, e.formatUser(user))
+func newRolePermissionPatch(action string, permission string, role string) ory.RelationshipPatch {
+	return ory.RelationshipPatch{
+		Action: &action,
+		RelationTuple: &ory.Relationship{
+			Namespace:  "Permission",
+			Object:     permission,
+			Relation:   "granted",
+			SubjectSet: ory.NewSubjectSet("Role", role, "member"),
+		},
 	}
+}
 
-	input := &models.OryAccessControlPolicy{
-		Actions:   actions,
-		Effect:    "allow",
-		ID:        fmtPolicy,
-		Resources: fmtResources,
-		Subjects:  append(fmtRoles, fmtUsers...),
-	}
-	params := &engines.UpsertOryAccessControlPolicyParams{
-		Body:   input,
-		Flavor: string(e.flavor),
-	}
-	res, err := e.ketoClient.UpsertOryAccessControlPolicy(params.WithTimeout(e.timeout))
-	if err != nil {
-		return nil, err
+func newRoleMemberPatch(action string, role string, member string) ory.RelationshipPatch {
+	return ory.RelationshipPatch{
+		Action: &action,
+		RelationTuple: &ory.Relationship{
+			Namespace:  "Role",
+			Object:     role,
+			Relation:   "member",
+			SubjectSet: ory.NewSubjectSet("Subject", member, ""),
+		},
 	}
-
-	payload := res.GetPayload()
-
-	return &types.Policy{
-		ID:        payload.ID,
-		Subjects:  payload.Subjects,
-		Resources: payload.Resources,
-		Actions:   payload.Actions,
-	}, nil
 }
 
-func (e *enforcer) isAllowed(user string, resource string, action string) (*bool, error) {
-	input := &models.OryAccessControlPolicyAllowedInput{
-		Action:   action,
-		Subject:  user,
-		Resource: resource,
+func (e *enforcer) UpdateAuthorization(ctx context.Context, updateRequest AuthorizationUpdateRequest) error {
```

Comment:
This function has complex logic. Would be good to have inline comments before each block, to summarize what's being done.